src/lib/storage/BaseRecord.ts

```diff
@@ -0,0 +1,35 @@
+export enum RecordType {
+  BaseRecord = 'BaseRecord',
+  ConnectionRecord = 'ConnectionRecord',
+  BasicMessageRecord = 'BasicMessageRecord',
+}
+
+export abstract class BaseRecord {
+  createdAt: number;
+  updatedAt?: number;
+  id: string;
+  tags: { [keys: string]: string };
+
+  // Required because Javascript doesn't allow accessing static types
+  // like instance.static_memeber
+  type: RecordType = RecordType.BaseRecord;
+  static type: RecordType = RecordType.BaseRecord;
+
+  constructor(id: string) {
+    this.id = id;
+    this.createdAt = new Date().getUTCMilliseconds();
```

Comment:
Is it correct Date function to use?  

From MDN docs: "The getUTCMilliseconds() method returns the milliseconds portion of the time object's value."

Maybe, the `Date.now()` is better. 
From MDN docs: "The Date.now() method returns the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC."

Reply:
my bad, I'll fix it